lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -148,21 +153,16 @@ def verify(creditcard, options = {})
         end
       end
 
-      def store(creditcard, options = {})
-        if options[:customer].present?
-          MultiResponse.new.tap do |r|
-            customer_exists_response = nil
-            r.process { customer_exists_response = check_customer_exists(options[:customer]) }
-            r.process do
-              if customer_exists_response.params['exists']
-                add_credit_card_to_customer(creditcard, options)
-              else
-                add_customer_with_credit_card(creditcard, options)
-              end
-            end
+      def store(payment_method, options = {})
+        return Response.new(false, bank_account_errors(payment_method, options)) if payment_method.is_a?(Check) && bank_account_errors(payment_method, options).present?
+        return add_customer_with_payment_method(payment_method, options) if options[:customer].blank?
+
+        MultiResponse.new.tap do |r|
+          r.process { check_customer_exists(options[:customer]) }
+          r.process do
+            process_by = r.params['exists'] ? :add_payment_method_to_customer : :add_customer_with_payment_method
+            send process_by, payment_method, options
```

Comment:
Hmm - I'm also not sure we want to nest logic within a multiresponse.
I also see that `MultiResponse` blocks are usually initiated with `.run` rather than `.new`. Do you have any thoughts on an alternative?

Reply:
@curiousepic on the .run vs .new.tap there is no difference .run is basically an alias for .new.tap, so I just change it for consistency.

For the usage of Multiresponse, I made a refactor to make the code clear and include the customer creation request in the multiresponse.